include/matx/transforms/matmul.h

```diff
@@ -1072,22 +1072,26 @@ void matmul(TensorTypeC C, const TensorTypeA A,
 {
   MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
 
+  // promote A and B to the type of C
+  auto A_ = as_type<typename TensorTypeC::scalar_type>(A);
```

Comment:
I don't think this will work for half precision mixed with full precision (yet). you could add a static assert to catch that and for now just handle the case where they're both fp32/fp64.

Reply:
Sorry but could you elaborate which mixed precision won't work? I am not sure about the same. Thanks

Reply:
Maybe it will now that I rethink that logic. It would have to be tested, but I'll close the comment for now.